### PR TITLE
Enable RISC-V 64-bit Cross-Compiling Support for ONNX Runtime on Linux

### PR DESCRIPTION
**Pull request by phoebesv**
### Description  
This pull request introduces the necessary changes to enable RISC-V 64-bit cross-compiling support for the ONNX Runtime on Linux. The RISC-V architecture has gained popularity as an open standard instruction set architecture, and this contribution aims to extend ONNX Runtime's compatibility to include RISC-V, thereby broadening the reach of ONNX models to a wider range of devices.

### Motivation and Context
RISC-V is a free and open-source instruction set architecture (ISA) based on established RISC principles. It is provided under open licenses without fees. Due to its extensibility and freedom in both software and hardware, RISC-V is poised for widespread adoption in the future, especially in applications related to AI, parallel computing, and data centers.

### Example Build Command
```
./build.sh --parallel --config Debug --rv64 --riscv_toolchain_root=/path/to/toolchain/root --skip_tests
```

### Documentation Updates
Relevant sections of the documentation will be updated to reflect the newly supported RISC-V 64-bit cross-compilation feature. https://github.com/microsoft/onnxruntime/pull/19239

### CODE REVIEW

**Comment by phoebesv** [comment]
@microsoft-github-policy-service agree company="SiFive"

**Review by snnn — APPROVED** [review]

File: cmake/riscv64.toolchain.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+# Copyright (c) 2024, Phoebe Chen <phoebe.chen@sifive.com>
+# Licensed under the MIT License.
+
+set(CMAKE_SYSTEM_NAME Linux)
+set(CMAKE_SYSTEM_PROCESSOR riscv64)
+
+list(APPEND CMAKE_TRY_COMPILE_PLATFORM_VARIABLES RISCV_TOOLCHAIN_ROOT)
+
+if(NOT RISCV_TOOLCHAIN_ROOT)
+  message(FATAL_ERROR "The option --riscv_toolchain_root is required.")
```
Comment (edgchen1) [inline cmake/riscv64.toolchain.cmake:11]: nit: can we avoid mentioning build.py options in CMake file error messages? some users may opt to use CMake directly.

**Comment by phoebesv** [comment]
@snnn I have updated the commits to fix the lint warning. Thanks!

**Review by snnn — APPROVED** [review]

**Comment by csukuangfj** [comment]
I am able to build onnxruntime for riscv64 with this PR. However, when I run it with my speech recognition model
using `qemu-riscv64`, it gives me wrong results.

Has the correctness of the compiled libs been verified/tested?

**Pull request merged**